### DISCUSSION TITLE
Getting coefficients/weights of added additional regressors

### DISCUSSION THREAD

**Discussion by PranavGuptInd** [ourownstory/neural_prophet#1162] [Q&A - get help using NeuralProphet] [answered]
Hi,

I have added some additional regressors to my neural prophet model by using model.add_future_regressor() as shown in documentation.
Now, I want to get the coefficients/weights of these regressors in the fitted model. I can see we can get these in plots using model.plot_parameters, but any other way we can get these values in a dataframe?

**Comment by noxan** [comment, marked as answer]
Hi @PranavGuptInd,
you can access the weights of regressors through the internal API `m.model.get_covar_weights(name)`.

```python
for name in m.config_regressors.keys():
    print(name, m.model.get_covar_weights(name))
```

Note: The name for the `config_regressors` might change in the future to `config_future_regressors` to be more precise.

Hope this helps and let us know if you have any other questions :)

**Reply by PranavGuptInd** [reply]
Thanks for answering. Also, can you please help me in getting the information of all the model parameters?
